cmd/orgs.go

```diff
@@ -215,6 +232,218 @@ func orgsRemove(ctx *cli.Context) error {
+		}
+	}
+
+	// Retrieve teams, memberships and the current session concurrently
+	var teams []envelope.Team
+	var memberships []envelope.Membership
+	var session *api.Session
+	var tErr, mErr, sErr error
+
+	var getMembersTeamsSession sync.WaitGroup
+	getMembersTeamsSession.Add(3)
+
+	go func() {
+		// Retrieve list of teams in org
+		teams, tErr = client.Teams.List(c, org.ID, "", primitive.AnyTeamType)
+		if tErr != nil {
+			tErr = errs.NewExitError("Could not retrieve list of teams.")
+			getMembersTeamsSession.Done()
+			return
+		}
+
+		getMembersTeamsSession.Done()
+	}()
+
+	go func() {
+		// Retrieve list of memberships in org
+		memberships, mErr = client.Memberships.List(c, org.ID, nil, nil)
+		if mErr != nil {
+			tErr = errs.NewExitError("Could not retrieve list of memberships.")
+			getMembersTeamsSession.Done()
+			return
+		}
+		getMembersTeamsSession.Done()
+	}()
+
+	go func() {
+		// Get current session - Who am I
+		session, err = client.Session.Who(c)
+		if sErr != nil {
+			sErr = errs.NewExitError("Failed to get current session.")
+			getMembersTeamsSession.Done()
+			return
+		}
+		getMembersTeamsSession.Done()
+	}()
+
+	getMembersTeamsSession.Wait()
+	if tErr != nil || mErr != nil || sErr != nil {
+		return cli.NewMultiError(
+			tErr,
+			mErr,
+			sErr,
+		)
+	}
+
+	if len(memberships) == 0 {
+		fmt.Printf("%s has no members\n", org.Body.Name)
+		return nil
+	}
+
+	// Map team IDs to Team objects
+	teamsIdx := make(map[identity.ID]envelope.Team)
+	for _, team := range teams {
+		teamsIdx[*team.ID] = team
+	}
+
+	userTeamIdx := make(map[identity.ID][]identity.ID) // Mapping from user ID to team IDs
+	membershipUserIDs := make(map[identity.ID]bool)    // Set of unique user IDs
+
+	// Create:
+	//	- Set of unqiue user IDs in membershipUserIDs
+	// 	- Mapping from user IDs to team IDs in userTeamIdx (1:m mapping)
+	for _, membership := range memberships {
+		// Skip memberships not associated with teams within org
+		team, ok := teamsIdx[*membership.Body.TeamID]
+		if !ok {
+			continue
+		}
+
+		// Skip memberships associated with machine team
+		if isMachineTeam(team.Body) {
+			continue
+		}
+
+		// Add to set of unique user IDs
+		membershipUserIDs[*membership.Body.OwnerID] = true
+
+		// For each new user ID, create mapping to teams that the user is in
+		_, ok = userTeamIdx[*membership.Body.OwnerID]
+		if !ok {
+			// For new user IDs (not yet in userTeamIdx) create an empty list of team IDs
+			userTeamIdx[*membership.Body.OwnerID] = []identity.ID{}
+		}
+		// Add current membership's team ID to user's list
+		userTeamIdx[*membership.Body.OwnerID] = append(userTeamIdx[*membership.Body.OwnerID], *membership.Body.TeamID)
+	}
+
+	// Create unique list of user IDs
+	var userIDs []identity.ID
+	for id := range membershipUserIDs {
+		userIDs = append(userIDs, id)
+	}
+
+	users, err := client.Profiles.ListByID(c, userIDs)
+	if err != nil {
+		return err
+	}
+	if users == nil {
+		return errs.NewExitError("User not found.")
```

Comment:
Some errors you start with capital and others you don't, we should probably standardize them (e.g. `org not found` vs. `User not found.`)